mmdet/models/dense_heads/corner_head.py

```diff
@@ -0,0 +1,1027 @@
+from math import ceil, log
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.cnn import ConvModule, bias_init_with_prob
+
+from mmdet.core import multi_apply
+from mmdet.ops import CornerPool, batched_nms
+from ..builder import HEADS, build_loss
+from ..utils import gaussian_radius, gen_gaussian_target
+
+
+class BiCornerPool(nn.Module):
+    """Bidirectional Corner Pooling Module (TopLeft, BottomRight, etc.)
+
+    Args:
+        in_channels (int): Input channels of module.
+        directions (list[str[): Directions of two CornerPools.
+        norm_cfg (dict): Dictionary to construct and config norm layer.
+    """
+
+    def __init__(self,
+                 in_channels,
+                 directions,
+                 feat_channels=128,
+                 norm_cfg=dict(type='BN', requires_grad=True)):
+        super(BiCornerPool, self).__init__()
+        self.direction1_conv = ConvModule(
+            in_channels, feat_channels, 3, padding=1, norm_cfg=norm_cfg)
+        self.direction2_conv = ConvModule(
+            in_channels, feat_channels, 3, padding=1, norm_cfg=norm_cfg)
+
+        self.aftpool_conv = ConvModule(
+            feat_channels,
+            in_channels,
+            3,
+            padding=1,
+            norm_cfg=norm_cfg,
+            act_cfg=None)
+
+        self.conv1 = ConvModule(
+            in_channels, in_channels, 1, norm_cfg=norm_cfg, act_cfg=None)
```

Comment:
This is a little bit wierd. `in_channels` should only be used for the input Conv, and `feat_channels` should be used for others. We may change `in_channels` to `feat_channels` here. If necessary, `out_channels` could be added.